soccer/optimization/GradientAscent1D.cpp

```diff
@@ -33,8 +31,8 @@ bool GradientAscent1D::singleStep() {
     currentdx = std::get<1>(funcOutput);
     currentVal = std::get<0>(funcOutput);
 
-    // Decrease temperature when derivative flips sign
-    if (sign(previousdx) == -1 * sign(currentdx)) {
+    // Decrease temperature when derivative flips signbit
+    if (signbit(previousdx) == -1 * signbit(currentdx)) {
```

Comment:
`signbit` returns a bool depending on whether the number is negative or not. I don't think this if statement has the same behavior that the original statement had. The original `sign` function returned `-1` when negative, `0` when 0, and `1` when positive.

Reply:
Yes you are totally right. I was just very lazy and hoped the bools would be auto-converted to ints and didnt test it. A find and replace wasn't quiet enough.

Reply:
FWIW, I probably would have done `signbit(previousdx) != signbit(currentdx)` to make it clear that we want to make sure the signs aren't equal.

This whole section of code optimizes a math function to find the local max. It's basically moves up the slope proportional to the steepness of the slope. The steeper it is, the further it moves. The extra piece here is that the derivative is not smooth so there are sharp corners. The sharp corners cause problems because the derivative is not guaranteed to go to zero. We deal with this by introducing the idea of temperature from simulated annealing. Every time the sign of the derivative changes (AKA we jump over one of these corners), we decrease the temperature. This temperature also controls how far it jumps. Eventually, after crossing a corner enough, the optimization "stops" at the top of the corner and returns the local max.